### PR TITLE
fix: Set '@typescript-eslint/return-await' rule to 'always' for FE (no-changelog)

### PR DESCRIPTION
**Pull request by tomi**
## Summary

Frontend follow-up of #8363

Returning a promise from function without awaiting it causes the function to be dropped from the stacktrace. This makes debugging harder. So this PR changes TS ESLint rule '@typescript-eslint/return-await' from its default value to 'always', so it is required to await any promises returned from async functions.

## Related tickets and issues

https://n8nio.slack.com/archives/C03MZF137FV/p1705496007084329

## Review / Merge checklist
- [x] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))

### CODE REVIEW

**Review by alexgrozav — CHANGES_REQUESTED** [review]
Looks good other than one small issue I found. Once that's fixed, we can move it further. 🚀

File: packages/@n8n/chat/src/api/generic.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export async function authenticatedFetch<T>(...args: Parameters<typeof fetch>):
 		},
 	});
 
-	return (await response.json()) as Promise<T>;
+	return await ((await response.json()) as Promise<T>);
```
[the remark this answers is not in the document]
Comment (tomi) [reply packages/@n8n/chat/src/api/generic.ts:19]: Oh nice catch! Will fix

**Review by alexgrozav — APPROVED** [review]
Thanks for addressing the issue!

**Comment by janober** [comment]
Got released with [`n8n@1.26.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.26.0)

**Pull request merged**